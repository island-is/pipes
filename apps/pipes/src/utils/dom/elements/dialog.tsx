import React from "react";

import { Text } from "../../ink/index.js";

import { Container } from "./container.js";
import { Row } from "./row.js";
import { Subtitle } from "./subtitle.js";

import type { Box } from "../../ink/index.js";
import type { ReactNode } from "react";

export type DialogType = "default" | "error" | "success" | "failure";
export type Color = Parameters<typeof Box>[0]["borderColor"];

export type DialogProps = {
  dialogType?: DialogType;
  children?: ReactNode | ReactNode[];
  title: string;
  paddingLeft?: number;
  paddingRight?: number;
  paddingTop?: number; // number of lines
  paddingBottom?: number; // number of lines
};
export type IDialog = { type: "Dialog" } & DialogProps;
export const Dialog = (props: Omit<IDialog, "type">): ReactNode => {
  const emojiType = {
    default: "",
    error: "X",
    success: "âœ”",
    failure: "!",
  }[props.dialogType ?? "default"];
  const borderColor = {
    default: "blue",
    error: "red",
    success: "green",
    failure: "red",
  }[props.dialogType ?? "default"];
  const allIsString = React.Children.toArray(props.children).every(
    (e) => typeof e === "string" || typeof e === "number",
  );
  let children = React.Children.map(React.Children.toArray(props.children), (e) => {
    if (allIsString) {
      return e;
    }
    if (typeof e === "string" || typeof e === "number") {
      return <Text>{`${e}`.trim()}</Text>;
    }
    return e;
  });
  if (allIsString) {
    children = [<Text key="text">{children.join("")}</Text>];
  }
  if (children.length === 0) {
    return <></>;
  }
  const dialogType = props.dialogType ?? "default";
  if (dialogType === "default") {
    return (
      <Row width={[1, undefined]}>
        <></>
        <Container>{children}</Container>
      </Row>
    );
  }
  return (
    <Row width={[18, undefined]}>
      <Subtitle color={borderColor} emoji={emojiType}>
        {props.title}
      </Subtitle>
      <Container>{children}</Container>
    </Row>
  );
};
