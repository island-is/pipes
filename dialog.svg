<svg aria-roledescription="flowchart-v2" role="graphics-document document" viewBox="-8 -8 442.1000061035156 572" style="max-width: 100%;" xmlns="http://www.w3.org/2000/svg" width="100%" id="graph-div" height="100%" xmlns:xlink="http://www.w3.org/1999/xlink"><style>@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css");'</style><style>#graph-div{font-family:"trebuchet ms",verdana,arial,sans-serif;font-size:16px;fill:#ccc;}#graph-div .error-icon{fill:#a44141;}#graph-div .error-text{fill:#ddd;stroke:#ddd;}#graph-div .edge-thickness-normal{stroke-width:2px;}#graph-div .edge-thickness-thick{stroke-width:3.5px;}#graph-div .edge-pattern-solid{stroke-dasharray:0;}#graph-div .edge-pattern-dashed{stroke-dasharray:3;}#graph-div .edge-pattern-dotted{stroke-dasharray:2;}#graph-div .marker{fill:lightgrey;stroke:lightgrey;}#graph-div .marker.cross{stroke:lightgrey;}#graph-div svg{font-family:"trebuchet ms",verdana,arial,sans-serif;font-size:16px;}#graph-div .label{font-family:"trebuchet ms",verdana,arial,sans-serif;color:#ccc;}#graph-div .cluster-label text{fill:#F9FFFE;}#graph-div .cluster-label span,#graph-div p{color:#F9FFFE;}#graph-div .label text,#graph-div span,#graph-div p{fill:#ccc;color:#ccc;}#graph-div .node rect,#graph-div .node circle,#graph-div .node ellipse,#graph-div .node polygon,#graph-div .node path{fill:#1f2020;stroke:#81B1DB;stroke-width:1px;}#graph-div .flowchart-label text{text-anchor:middle;}#graph-div .node .label{text-align:center;}#graph-div .node.clickable{cursor:pointer;}#graph-div .arrowheadPath{fill:lightgrey;}#graph-div .edgePath .path{stroke:lightgrey;stroke-width:2.0px;}#graph-div .flowchart-link{stroke:lightgrey;fill:none;}#graph-div .edgeLabel{background-color:hsl(0, 0%, 34.4117647059%);text-align:center;}#graph-div .edgeLabel rect{opacity:0.5;background-color:hsl(0, 0%, 34.4117647059%);fill:hsl(0, 0%, 34.4117647059%);}#graph-div .labelBkg{background-color:rgba(87.75, 87.75, 87.75, 0.5);}#graph-div .cluster rect{fill:hsl(180, 1.5873015873%, 28.3529411765%);stroke:rgba(255, 255, 255, 0.25);stroke-width:1px;}#graph-div .cluster text{fill:#F9FFFE;}#graph-div .cluster span,#graph-div p{color:#F9FFFE;}#graph-div div.mermaidTooltip{position:absolute;text-align:center;max-width:200px;padding:2px;font-family:"trebuchet ms",verdana,arial,sans-serif;font-size:12px;background:hsl(20, 1.5873015873%, 12.3529411765%);border:1px solid rgba(255, 255, 255, 0.25);border-radius:2px;pointer-events:none;z-index:100;}#graph-div .flowchartTitleText{text-anchor:middle;font-size:18px;fill:#ccc;}#graph-div :root{--mermaid-font-family:"trebuchet ms",verdana,arial,sans-serif;}</style><g><marker orient="auto" markerHeight="12" markerWidth="12" markerUnits="userSpaceOnUse" refY="5" refX="10" viewBox="0 0 10 10" class="marker flowchart" id="flowchart-pointEnd"><path style="stroke-width: 1px; stroke-dasharray: 1px, 0px;" class="arrowMarkerPath" d="M 0 0 L 10 5 L 0 10 z"></path></marker><marker orient="auto" markerHeight="12" markerWidth="12" markerUnits="userSpaceOnUse" refY="5" refX="0" viewBox="0 0 10 10" class="marker flowchart" id="flowchart-pointStart"><path style="stroke-width: 1px; stroke-dasharray: 1px, 0px;" class="arrowMarkerPath" d="M 0 5 L 10 10 L 10 0 z"></path></marker><marker orient="auto" markerHeight="11" markerWidth="11" markerUnits="userSpaceOnUse" refY="5" refX="11" viewBox="0 0 10 10" class="marker flowchart" id="flowchart-circleEnd"><circle style="stroke-width: 1px; stroke-dasharray: 1px, 0px;" class="arrowMarkerPath" r="5" cy="5" cx="5"></circle></marker><marker orient="auto" markerHeight="11" markerWidth="11" markerUnits="userSpaceOnUse" refY="5" refX="-1" viewBox="0 0 10 10" class="marker flowchart" id="flowchart-circleStart"><circle style="stroke-width: 1px; stroke-dasharray: 1px, 0px;" class="arrowMarkerPath" r="5" cy="5" cx="5"></circle></marker><marker orient="auto" markerHeight="11" markerWidth="11" markerUnits="userSpaceOnUse" refY="5.2" refX="12" viewBox="0 0 11 11" class="marker cross flowchart" id="flowchart-crossEnd"><path style="stroke-width: 2px; stroke-dasharray: 1px, 0px;" class="arrowMarkerPath" d="M 1,1 l 9,9 M 10,1 l -9,9"></path></marker><marker orient="auto" markerHeight="11" markerWidth="11" markerUnits="userSpaceOnUse" refY="5.2" refX="-1" viewBox="0 0 11 11" class="marker cross flowchart" id="flowchart-crossStart"><path style="stroke-width: 2px; stroke-dasharray: 1px, 0px;" class="arrowMarkerPath" d="M 1,1 l 9,9 M 10,1 l -9,9"></path></marker><g class="root"><g class="clusters"></g><g class="edgePaths"><path marker-end="url(#flowchart-pointEnd)" marker-start="url(#flowchart-pointStart)" style="fill:none;" class="edge-thickness-normal edge-pattern-solid flowchart-link LS-A LE-B" id="L-A-B-0" d="M213.0500030517578,39L213.0500030517578,45.166666666666664C213.0500030517578,51.333333333333336,213.0500030517578,63.666666666666664,213.0500030517578,76C213.0500030517578,88.33333333333333,213.0500030517578,100.66666666666667,213.0500030517578,106.83333333333333L213.0500030517578,113"></path><path marker-end="url(#flowchart-pointEnd)" marker-start="url(#flowchart-pointStart)" style="fill:none;" class="edge-thickness-normal edge-pattern-solid flowchart-link LS-B LE-C" id="L-B-C-0" d="M213.0500030517578,152L213.0500030517578,158.16666666666666C213.0500030517578,164.33333333333334,213.0500030517578,176.66666666666666,213.0500030517578,189C213.0500030517578,201.33333333333334,213.0500030517578,213.66666666666666,213.0500030517578,219.83333333333334L213.0500030517578,226"></path><path marker-end="url(#flowchart-pointEnd)" marker-start="url(#flowchart-pointStart)" style="fill:none;" class="edge-thickness-normal edge-pattern-solid flowchart-link LS-C LE-D" id="L-C-D-0" d="M213.0500030517578,265L213.0500030517578,269.1666666666667C213.0500030517578,273.3333333333333,213.0500030517578,281.6666666666667,213.0500030517578,290C213.0500030517578,298.3333333333333,213.0500030517578,306.6666666666667,213.0500030517578,310.8333333333333L213.0500030517578,315"></path><path marker-end="url(#flowchart-pointEnd)" marker-start="url(#flowchart-pointStart)" style="fill:none;" class="edge-thickness-normal edge-pattern-solid flowchart-link LS-D LE-E" id="L-D-E-0" d="M213.0500030517578,354L213.0500030517578,360.1666666666667C213.0500030517578,366.3333333333333,213.0500030517578,378.6666666666667,213.0500030517578,391C213.0500030517578,403.3333333333333,213.0500030517578,415.6666666666667,213.0500030517578,421.8333333333333L213.0500030517578,428"></path><path marker-end="url(#flowchart-pointEnd)" style="fill:none;" class="edge-thickness-normal edge-pattern-solid flowchart-link LS-E LE-F" id="L-E-F-0" d="M213.0500030517578,467L213.0500030517578,473.1666666666667C213.0500030517578,479.3333333333333,213.0500030517578,491.6666666666667,213.0500030517578,504C213.0500030517578,516.3333333333334,213.0500030517578,528.6666666666666,213.0500030517578,534.8333333333334L213.0500030517578,541"></path></g><g class="edgeLabels"><g transform="translate(213.0500030517578, 76)" class="edgeLabel"><g transform="translate(-70.69999694824219, -12)" class="label"><foreignObject height="24" width="141.39999389648438"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel">written in Typescript</span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 189)" class="edgeLabel"><g transform="translate(-140.56666564941406, -12)" class="label"><foreignObject height="24" width="281.1333312988281"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel">Zod parsers input/output context, config</span></div></foreignObject></g></g><g class="edgeLabel"><g transform="translate(0, 0)" class="label"><foreignObject height="0" width="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 391)" class="edgeLabel"><g transform="translate(-146.7833251953125, -12)" class="label"><foreignObject height="24" width="293.566650390625"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel">Mobx saves states, delegates to modules</span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 504)" class="edgeLabel"><g transform="translate(-213.0500030517578, -12)" class="label"><foreignObject height="24" width="426.1000061035156"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel">Dagger manages building/test etc. using buildkit and Docker</span></div></foreignObject></g></g></g><g class="nodes"><g transform="translate(213.0500030517578, 19.5)" id="flowchart-A-1369" class="node default default flowchart-label"><rect height="39" width="138.93333435058594" y="-19.5" x="-69.46666717529297" ry="0" rx="0" style="" class="basic label-container"></rect><g transform="translate(-61.96666717529297, -12)" style="" class="label"><rect></rect><foreignObject height="24" width="123.93333435058594"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">Your CI/CID code</span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 132.5)" id="flowchart-B-1370" class="node default default flowchart-label"><rect height="39" width="231.96665954589844" y="-19.5" x="-115.98332977294922" ry="0" rx="0" style="" class="basic label-container"></rect><g transform="translate(-108.48332977294922, -12)" style="" class="label"><rect></rect><foreignObject height="24" width="216.96665954589844"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">Compiled with SWC in runtime</span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 245.5)" id="flowchart-C-1372" class="node default default flowchart-label"><rect height="39" width="119.94999694824219" y="-19.5" x="-59.974998474121094" ry="0" rx="0" style="" class="basic label-container"></rect><g transform="translate(-52.474998474121094, -12)" style="" class="label"><rect></rect><foreignObject height="24" width="104.94999694824219"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">Pipes modules</span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 334.5)" id="flowchart-D-1374" class="node default default flowchart-label"><rect height="39" width="94.14999389648438" y="-19.5" x="-47.07499694824219" ry="0" rx="0" style="" class="basic label-container"></rect><g transform="translate(-39.57499694824219, -12)" style="" class="label"><rect></rect><foreignObject height="24" width="79.14999389648438"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">Pipes Core</span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 447.5)" id="flowchart-E-1376" class="node default default flowchart-label"><rect height="39" width="67.48333740234375" y="-19.5" x="-33.741668701171875" ry="0" rx="0" style="" class="basic label-container"></rect><g transform="translate(-26.241668701171875, -12)" style="" class="label"><rect></rect><foreignObject height="24" width="52.48333740234375"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">Dagger</span></div></foreignObject></g></g><g transform="translate(213.0500030517578, 548.5)" id="flowchart-F-1378" class="node default default flowchart-label"><rect height="15" width="15" y="-7.5" x="-7.5" ry="0" rx="0" style="" class="basic label-container"></rect><g transform="translate(0, 0)" style="" class="label"><rect></rect><foreignObject height="0" width="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel"></span></div></foreignObject></g></g></g></g></g></svg>