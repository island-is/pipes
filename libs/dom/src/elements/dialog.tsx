import { Box, Text } from "@island-is/ink";
import React from "react";

import { Subtitle } from "./subtitle.js";

import type { ReactNode } from "react";

export type DialogType = "default" | "error" | "success" | "failure";
export type Color = Parameters<typeof Box>[0]["borderColor"];

export type DialogProps = {
  dialogType?: DialogType;
  children?: ReactNode | ReactNode[];
  title: string;
  paddingLeft?: number;
  paddingRight?: number;
  paddingTop?: number; // number of lines
  paddingBottom?: number; // number of lines
};
export type IDialog = { type: "Dialog" } & DialogProps;
export const Dialog = (props: Omit<IDialog, "type">): ReactNode => {
  const emojiType = {
    default: "*",
    error: "X",
    success: "âœ”",
    failure: "!",
  }[props.dialogType ?? "default"];
  const borderColor = {
    default: "blue",
    error: "red",
    success: "green",
    failure: "red",
  }[props.dialogType ?? "default"];
  const children = React.Children.map(React.Children.toArray(props.children), (e) => {
    if (typeof e === "string" || typeof e === "number") {
      return <Text>{`${e}`.trim()}</Text>;
    }
    return e;
  });

  return (
    <Box
      borderStyle={"double"}
      borderColor={borderColor}
      flexDirection="column"
      width={"100%"}
      paddingLeft={props.paddingLeft ?? 1}
      paddingRight={props.paddingLeft ?? 1}
      paddingTop={props.paddingTop ?? 1}
      paddingBottom={props.paddingBottom ?? 1}
    >
      <Subtitle color={borderColor} emoji={emojiType}>
        {props.title}
      </Subtitle>
      <Box width="100%" height={(props.paddingBottom ?? 1) + (props.paddingTop ?? 1)}></Box>
      {children}
    </Box>
  );
};
