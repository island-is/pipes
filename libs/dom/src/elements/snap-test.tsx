import assert from "node:assert";
import { basename, dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

import { SnapshotState } from "jest-snapshot";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
function toMatchSnapshot(actual: any, testFile: string, testTitle: string) {
  const root = join(fileURLToPath(dirname(testFile)), "__snapshots__");
  const fileName = `${basename(fileURLToPath(testFile))}.snap`;
  const snapshotFile = join(root, fileName);
  const snapshotState = new SnapshotState(snapshotFile, {
    updateSnapshot: process.env.SNAPSHOT_UPDATE ? "all" : "new",
  } as any);

  const matcher = snapshotState.match({
    testName: `${testTitle}-${testFile}`,
    received: actual,
    isInline: false,
  });

  const result = matcher;
  snapshotState.save();

  return result;
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const snapTest = (actual: any, testFile: string, testTitle: string) => {
  const result = toMatchSnapshot(actual, testFile, testTitle);
  assert(result.pass, `${testTitle} - does not match snapshot`);
};
