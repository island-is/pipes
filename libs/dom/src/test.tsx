import { consoleRender } from "./ci.js";
import { PipesDOM } from "./dom.js";
import { summaryRender } from "./markdown.js";

const dashboard2 = (
  <PipesDOM.Group title="Test">
    <PipesDOM.Mask values={["Dashboard"]} />
    <PipesDOM.Container>
      <PipesDOM.Highlight>Note: User feedback will be available tomorrow.</PipesDOM.Highlight>
      <PipesDOM.Dialog dialogType="error" message="There was an error processing your request. WOWOWOWOW!" />
      <PipesDOM.Title>Dashboard Overview</PipesDOM.Title>
      <PipesDOM.Subtitle>Metrics for August 2023</PipesDOM.Subtitle>
      <PipesDOM.Success>Data fetched successfully!</PipesDOM.Success>
      <PipesDOM.Timestamp time={"12.22.2020"} />
      <PipesDOM.Failure>Failed fetching data!</PipesDOM.Failure>
      <PipesDOM.Warning>There was a slight discrepancy in the financial data. Please review.</PipesDOM.Warning>
      <PipesDOM.Table>
        <PipesDOM.TableHeadings>
          <PipesDOM.TableCell>Metric</PipesDOM.TableCell>
          <PipesDOM.TableCell>Value</PipesDOM.TableCell>
          <PipesDOM.TableCell>Change since last month</PipesDOM.TableCell>
        </PipesDOM.TableHeadings>
        <PipesDOM.TableRow>
          <PipesDOM.TableCell>Users</PipesDOM.TableCell>
          <PipesDOM.TableCell>4,500</PipesDOM.TableCell>
          <PipesDOM.TableCell>+15%</PipesDOM.TableCell>
        </PipesDOM.TableRow>
        <PipesDOM.TableRow>
          <PipesDOM.TableCell>Revenue</PipesDOM.TableCell>
          <PipesDOM.TableCell>$12,000</PipesDOM.TableCell>
          <PipesDOM.TableCell>-2%</PipesDOM.TableCell>
        </PipesDOM.TableRow>
        <PipesDOM.TableRow>
          <PipesDOM.TableCell>Subscriptions</PipesDOM.TableCell>
          <PipesDOM.TableCell>2,000</PipesDOM.TableCell>
          <PipesDOM.TableCell>+8%</PipesDOM.TableCell>
        </PipesDOM.TableRow>
      </PipesDOM.Table>
      <PipesDOM.List>
        <PipesDOM.ListItem>Tasks completed: 75</PipesDOM.ListItem>
        <PipesDOM.ListItem>Pending tasks: 12</PipesDOM.ListItem>
        <PipesDOM.ListItem>Overdue tasks: 3</PipesDOM.ListItem>
      </PipesDOM.List>
      <PipesDOM.Error>Failed to fetch advertisement data. Please check the ad server.</PipesDOM.Error>
      <PipesDOM.Highlight>Note: User feedback will be available tomorrow.</PipesDOM.Highlight>
      <PipesDOM.Divider />
      <PipesDOM.Link href="https://www.mbl.is">mbl.is</PipesDOM.Link>
      <PipesDOM.Row>
        <PipesDOM.Badge>Admin</PipesDOM.Badge>
        <PipesDOM.Text value="Logged in as: JohnDoe" />
      </PipesDOM.Row>
    </PipesDOM.Container>
  </PipesDOM.Group>
);

const summary = (
  <PipesDOM.Container>
    <PipesDOM.Title>Wow</PipesDOM.Title>
    <PipesDOM.Table>
      <PipesDOM.TableHeadings>
        <PipesDOM.TableCell>Metric</PipesDOM.TableCell>
        <PipesDOM.TableCell>Value</PipesDOM.TableCell>
        <PipesDOM.TableCell>Change since last month</PipesDOM.TableCell>
      </PipesDOM.TableHeadings>
      <PipesDOM.TableRow>
        <PipesDOM.TableCell>Users</PipesDOM.TableCell>
        <PipesDOM.TableCell>4,500</PipesDOM.TableCell>
        <PipesDOM.TableCell>+15%</PipesDOM.TableCell>
      </PipesDOM.TableRow>
      <PipesDOM.TableRow>
        <PipesDOM.TableCell>Revenue</PipesDOM.TableCell>
        <PipesDOM.TableCell>$12,000</PipesDOM.TableCell>
        <PipesDOM.TableCell>-2%</PipesDOM.TableCell>
      </PipesDOM.TableRow>
      <PipesDOM.TableRow>
        <PipesDOM.TableCell>Subscriptions</PipesDOM.TableCell>
        <PipesDOM.TableCell>2,000</PipesDOM.TableCell>
        <PipesDOM.TableCell>+8%</PipesDOM.TableCell>
      </PipesDOM.TableRow>
    </PipesDOM.Table>
  </PipesDOM.Container>
);

// Now you can use this to render to the terminal:
consoleRender.mount(dashboard2);
await consoleRender.render();
summaryRender.mount(summary);
await summaryRender.render();
