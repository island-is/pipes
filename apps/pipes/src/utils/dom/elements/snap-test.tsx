import assert from "node:assert";
import { basename, dirname, join } from "node:path";
import { it } from "node:test";
import { fileURLToPath } from "node:url";

import { SnapshotState } from "jest-snapshot";

import render from "../../ink/render.js";

import type { ReactNode } from "react";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
function toMatchSnapshot(actual: any, testFile: string, testTitle: string) {
  const root = join(fileURLToPath(dirname(testFile)), "__snapshots__");
  const fileName = `${basename(fileURLToPath(testFile))}.snap`;
  const snapshotFile = join(root, fileName);
  console.log(snapshotFile);
  const snapshotState = new SnapshotState(snapshotFile, {
    updateSnapshot: process.env.SNAPSHOT_UPDATE ? "all" : "new",
  } as any);

  const matcher = snapshotState.match({
    testName: `${testTitle}-${testFile}`,
    received: actual,
    isInline: false,
  });

  const result = matcher;
  snapshotState.save();

  return result;
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const snapTest = (actual: any, testFile: string, testTitle: string) => {
  const result = toMatchSnapshot(actual, testFile, testTitle);
  assert(result.pass, `${testTitle} ${basename(testFile)} - does not match snapshot`);
};

export const testJSX = async (jsx: ReactNode, name: string, filename: string): Promise<void> => {
  await it(name, async () => {
    const rendered = await render(jsx, true);
    const value = rendered.value();
    console.log(`${name}\n${value}`);
    snapTest(value, filename, name);
  });
};
