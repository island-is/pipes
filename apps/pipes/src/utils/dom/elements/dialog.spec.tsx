import { describe } from "node:test";

import React from "react";

import { Dialog } from "./dialog.js";
import { LOREM_IPSUM } from "./lorem.ipsum.spec.js";
import { testJSX } from "./snap-test.js";

describe("Render dialog", () => {
  describe("ansi", async () => {
    const tests = [
      testJSX(<Dialog title="Test">This is a message</Dialog>, "render", import.meta.url),
      testJSX(
        <Dialog title="Test" paddingBottom={1} paddingTop={1}>
          This is a message
        </Dialog>,
        "padding",
        import.meta.url,
      ),
      testJSX(
        <Dialog title="Test" paddingBottom={1} paddingTop={1}>
          {LOREM_IPSUM}
        </Dialog>,
        "multiple lines",
        import.meta.url,
      ),
      testJSX(
        <Dialog title="Test" dialogType={"error"} paddingBottom={1} paddingTop={1}>
          {LOREM_IPSUM}
        </Dialog>,
        "error",
        import.meta.url,
      ),
    ];
    await Promise.all(tests);
  });
});
