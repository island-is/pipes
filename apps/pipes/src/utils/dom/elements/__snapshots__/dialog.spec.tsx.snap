// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error 1`] = `""`;

exports[`multiple lines 1`] = `""`;

exports[`padding 1`] = `""`;

exports[`render 1`] = `""`;
